main.c

```diff
@@ -14,12 +14,13 @@ int main() {
     set_piezo_irq();
 
     while (true) {
-        if (state.current_comp_idx == PILLCOMP_COUNT) {
+        if (!state.dispensing_underway) {
             blink_until_sw_pressed(&state);
         }
 
         calibrate(&state);
-        if (state.current_comp_idx == PILLCOMP_COUNT) {
+
+        if (!state.dispensing_underway) {
```

Comment:
Ye, I thought about that...
It does leave a subtle (and small) hole in the logic, though:
If the program is rebooted after having started dispensing but not finishing the very first rotation,
it would wait for the button press, because comp_idx == 0 ; however, we'd want to skip it in this case.

While it is an extra variable, I do think it makes the code more readable.
```if (comp_idx == PILLCOMP)``` ... "Why?" would be my question, if I was reviewing it.

Reply:
okay. well explained. I didnt think abt the case when reboot happens during first dispensing. Let's keep it that way